torch_geometric/nn/norm/agg_norm.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+from torch import Tensor
+
+from torch_geometric.nn.aggr import Aggregation
+from torch_geometric.nn.resolver import aggregation_resolver
+
+
+class AggSubtraction(torch.nn.Module):
+    r"""Applies layer normalization by subtracting an aggregation
+    over the inputs. Supports providing an aggregator from the
+    :class:`~torch_geometric.nn.aggr.Aggregation` module.
+
+    Args:
+        aggr (string or list or Aggregation, optional): The aggregation scheme
+            to use, *e.g.*, :obj:`"add"`, :obj:`"sum"` :obj:`"mean"`,
+            :obj:`"min"`, :obj:`"max"` or :obj:`"mul"`.
+            In addition, can be any
+            :class:`~torch_geometric.nn.aggr.Aggregation` module (or any string
+            that automatically resolves to it).
+            If given as a list, will make use of multiple aggregations in which
+            different outputs will get concatenated in the last dimension.
+            If set to :obj:`None`, the :class:`MessagePassing` instantiation is
+            expected to implement its own aggregation logic via
+            :meth:`aggregate`. (default: :obj:`"mean"`)
+        aggr_kwargs (Dict[str, Any], optional): Arguments passed to the
+            respective aggregation function in case it gets automatically
+            resolved. (default: :obj:`None`)
+    """
+    def __init__(
+        self,
+        aggr: Optional[Union[str, List[str], Aggregation]] = "mean",
+        aggr_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        super().__init__()
+        self.aggr_module = aggregation_resolver(aggr, **(aggr_kwargs or {}))
+
+    def forward(self, x: Tensor, index: Optional[Tensor] = None):
+        agg = self.aggr_module(x, index)
```

Comment:
Will this lead to suboptimal implementation if `index` is absent? Should we use dense reduce `torch.mean()` when `index` is absent?

Reply:
It will, but I was wondering if we should fix this inside of the Àggregation` implementation - WDTY?

Reply:
Does it even make sense to "fix" inside `Aggregation` actually? Maybe I'm wrong here.

Reply:
Mh, supporting this in `Aggregation` is tricky since we will need to implement a dense reduction for every new aggregator with little to no use. Let‘s add this to the norm layer for now until we have a clear understanding on how to implement that within `Aggregation`.

Reply:
Agree. Fixing it in the norm layer should be enough for now.

Reply:
Alright fair enough.

My thought was we don't need to support dense for all aggregations, but I agree we can at least address it elsewhere.

Reply:
I reverted to the original implementation: Could have used `global_mean_pool` but we've gotten ourselves into a bit of a twist with that package and some circular imports. In this case because pool itself is importing other modules.

We  can leave this here, but we might want to add a solution to the nn.aggr roadmap.